parsers/cpreprocessor.c

```diff
@@ -518,6 +552,7 @@ static int cppGetcFromUngetBufferOrFile(void)
 		return c;
 	}
 
+	cppClearMacroInUse (&Cpp.macroInUse);
```

Comment:
This call is in a very hot path.
I suggest to moving it to cppUngetStringBuiltByMacro() like this:

```
if(!Cpp.ungetPointer)
   cppClearMacroInUse(&Cpp.macroInUse)
```
I know that this is imperfect because it will retain macro usage for one substitution AFTER a sequence. However this should not hurt as long as we keep the maximum number of macro usages in a sequence high (See other comment below).

If you have a better idea for a place to move it we can use yours - the point is to move it out of the hot path.

Reply:
Cpp.ungetPointer? Didn't you want to write "Cpp.macroInUse"?

We can expect Cpp.ungetPointer is always NULL because (1) the if-condition before the line I changed is 
`if (Cpp.ungetPointer)` and (2) `return` statement exists at the end of the if-block.
So the change you suggested doesn't change the control flow. Am I missing something?

Reply:
I used "Cpp.macroInUse" as the condition in the update.

Reply:
The current version is quite better as it avoids the function call in most cases. So we can leave it like this.


Just for clarification of my previous comment, I suggested to moving the macro cleaning call to a different function: cppUngetStringBuiltByMacro().

So instead of 
```
static int cppGetcFromUngetBufferOrFile(void)
{
	...
	if (Cpp.macroInUse)
		cppClearMacroInUse (&Cpp.macroInUse);
	...
}
```
which is in the "hot path" because is called for every character read, do
```
extern void cppUngetStringBuiltByMacro(const char * string,int len, cppMacroInfo *macro)
{
	if (!Cpp.ungetPointer)
		// first macro replacement in sequence
		cppClearMacroInUse (&Cpp.macroInUse);
	...
}
```
which is not in the "hot path".

This is imperfect since the first replacement in a sequence retains the macro counts from the previous sequence. But as long as the limit is high, it won't hurt.

Though in fact with your additional check the difference might be negligible.

Reply:
Oh, I see. However, I would like to keep this pull request simple. So I would like you to create a new pull request stacked on my change (if you want).